File: querybean-generator/src/test/java/io/ebean/querybean/generator/UtilTest.java
package io.ebean.querybean.generator;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class UtilTest {

  @Test
  void stripForMethod() {
    assertEquals(Util.stripForMethod("-a"), "_a");
    assertEquals(Util.stripForMethod("-a-b-"), "_a_b_");
    assertEquals(Util.stripForMethod("c-"), "c_");
    assertEquals(Util.stripForMethod("my-foo"), "my_foo");
  }

  @Test
  void packageOf() {
    assertEquals(Util.packageOf(true, "com.example.Foo.Bar"), "com.example");
    assertEquals(Util.packageOf(false, "com.example.other.foo.Bar"), "com.example.other.foo");
  }

  @Test
  void shortName() {
    assertEquals(Util.shortName(true, "com.example.Foo.Bar"), "Foo.Bar");
    assertEquals(Util.shortName(false, "com.example.foo.Bar"), "Bar");
    assertEquals(Util.shortName(false, "Bar"), "Bar");
  }

}

File: kotlin-querybean-generator/src/test/kotlin/io/ebean/querybean/generator/UtilTest.kt
package io.ebean.querybean.generator

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class UtilTest {

  @Test
  fun stripForMethod() {
    Assertions.assertEquals(io.ebean.querybean.generator.Util.stripForMethod("-a"), "_a")
    Assertions.assertEquals(io.ebean.querybean.generator.Util.stripForMethod("-a-b-"), "_a_b_")
    Assertions.assertEquals(io.ebean.querybean.generator.Util.stripForMethod("c-"), "c_")
    Assertions.assertEquals(io.ebean.querybean.generator.Util.stripForMethod("my-foo"), "my_foo")
  }

  @Test
  fun packageOf() {
    Assertions.assertEquals(Util.packageOf(true, "com.example.Foo.Bar"), "com.example")
    Assertions.assertEquals(Util.packageOf(false, "com.example.other.foo.Bar"), "com.example.other.foo")
  }

  @Test
  fun shortName() {
    Assertions.assertEquals(Util.shortName(true, "com.example.Foo.Bar"), "Foo.Bar")
    Assertions.assertEquals(Util.shortName(false, "com.example.foo.Bar"), "Bar")
    Assertions.assertEquals(Util.shortName(false, "Bar"), "Bar")
  }
}
